flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/config/MongoReadOptions.java

```diff
@@ -0,0 +1,230 @@
+        return noCursorTimeout;
+    }
+
+    public PartitionStrategy getPartitionStrategy() {
+        return partitionStrategy;
+    }
+
+    public MemorySize getPartitionSize() {
+        return partitionSize;
+    }
+
+    public int getSamplesPerPartition() {
+        return samplesPerPartition;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        MongoReadOptions that = (MongoReadOptions) o;
+        return cursorBatchSize == that.cursorBatchSize
+                && noCursorTimeout == that.noCursorTimeout
+                && partitionStrategy == that.partitionStrategy
+                && samplesPerPartition == that.samplesPerPartition
+                && Objects.equals(partitionSize, that.partitionSize);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(
+                cursorBatchSize,
+                noCursorTimeout,
+                partitionStrategy,
+                partitionSize,
+                samplesPerPartition);
+    }
+
+    public static MongoReadOptionsBuilder builder() {
+        return new MongoReadOptionsBuilder();
+    }
+
+    /** Builder for {@link MongoReadOptions}. */
+    @PublicEvolving
+    public static class MongoReadOptionsBuilder {
+        private int fetchSize = SCAN_FETCH_SIZE.defaultValue();
+        private int cursorBatchSize = SCAN_CURSOR_BATCH_SIZE.defaultValue();
+        private boolean noCursorTimeout = SCAN_CURSOR_NO_TIMEOUT.defaultValue();
+        private PartitionStrategy partitionStrategy = SCAN_PARTITION_STRATEGY.defaultValue();
+        private MemorySize partitionSize = SCAN_PARTITION_SIZE.defaultValue();
+        private int samplesPerPartition = SCAN_PARTITION_SAMPLES.defaultValue();
+
+        /**
+         * Sets the number of documents should be fetched per round-trip when reading.
+         *
+         * @param fetchSize the number of documents should be fetched per round-trip when reading.
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setFetchSize(int fetchSize) {
+            checkArgument(fetchSize > 0, "The fetch size must be larger than 0.");
+            this.fetchSize = fetchSize;
+            return this;
+        }
+
+        /**
+         * Sets the batch size of MongoDB find cursor.
+         *
+         * @param cursorBatchSize the max batch size of find cursor.
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setCursorBatchSize(int cursorBatchSize) {
+            checkArgument(
+                    cursorBatchSize >= 0,
+                    "The cursor batch size must be larger than or equal to 0.");
+            this.cursorBatchSize = cursorBatchSize;
+            return this;
+        }
+
+        /**
+         * Set this option to true to prevent cursor timeout (10 minutes).
```

Comment:
The MongoDB server normally times out idle cursors after an inactivity period (10minutes) to prevent excess memory use. 

Refers to https://www.mongodb.com/docs/manual/reference/method/cursor.noCursorTimeout/#session-idle-timeout-overrides-nocursortimeout
```
If a session is idle for longer than 30 minutes, the MongoDB server marks that session as expired 
and may close it at any time. 
When the MongoDB server closes the session, it also kills any in-progress operations and open 
cursors associated with the session. 
This includes cursors configured with noCursorTimeout() or a maxTimeMS() greater than 30 minutes.
```

We can use this option to make the cursor live longer than 10 minutes, but it will be less than 30 minutes.
This is useful for some scenarios where the query takes a long time.